### ISSUE TITLE
Blank screen after wake from sleep after the recent 4K/DP related changes

### ISSUE THREAD

**Issue by nijhawank** [MSzturc/Lenovo-T460-OpenCore#8]
On my T460, I’m having blank screen after wake from sleep after the recent 4K/DP related changes. It does wake from sleep and the screen is backlit but the screen is blank. 
After I reverted the 4K/DP changes, the wake from sleep is back to normal and working.

**Comment by MSzturc** [comment]
Are you using an external Monitor?

**Comment by nijhawank** [comment]
No, just the laptop screen itself. On some checking, dpcd-max-link-rate may be the problem. The value that you have specified is 0x14 which is for 4K display while my screen is 1080p and the value should be 0x0A (per WhateverGreen documentation). I haven't verified it yet however. Is your screen 4k?

**Comment by MSzturc** [comment]
This change was needed because i've added an external monitor to my daily use setup. Its an 1440p Monitor with 144hz (which uses a similar clockrate as an 4k monitor with 60hz). To support the higher clockrate the dpcd-max-link-rate was needed. In this mode the internal monitor gets deactivated as soon as i connect an external monitor via dp. Currently there is a bug that the internal monitor doesnt activates after the internal was connected once (a reboot is needed). But when I boot without external monitor the config works as before.

**Comment by MSzturc** [comment]
Today i've patched my bios. With modified bios the patch is not longer needed to initialize monitor, so i reverted it, Please test if your problem still occur with current version

**Comment by simprecicchiani** [comment]
> Today i've patched my bios. With modified bios the patch is not longer needed to initialize monitor, so i reverted it, Please test if your problem still occur with current version

Hi, would you recommend patching the bios? Are you able to access DVMT and CFG Lock via the Advanced tab? I didn't find anyone who did it on 6th gen Thinkpad.

**Comment by MSzturc** [comment]
I followed @tylernguyen guide [here](https://github.com/tylernguyen/x1c6-hackintosh/blob/master/docs/1_README-HARDWAREandBIOS.md). His x1c6 is based on Skylake CPU so i gave it a try. But to be honest, my fingers where a bit sweaty when i patched the bios the first time.

**Comment by simprecicchiani** [comment]
Yeah I saw the guide and I can image your concerns while doing it. I'm planning to do it myself as soon as I get my other Hackintosh desktop back.
I wanted to know if you experienced a bump in performance, thermals, stability, startup time or something...

**Comment by MSzturc** [comment]
To be honest, no, but since i patched it i was busy in other projects, the next weekend im alone i will spend some time fiddling with it

**Comment by curiexr** [comment]
Hi, are you still using the external monitor and if so, how is the performance? I have a 4K monitor and would like to get 4k/60Hz working (and disabling the internal monitor while the external is connected). Is patching bios necessary to make this work? Thank you!
